ecl/eclscheduler/eclscheduler.cpp

```diff
@@ -172,14 +154,200 @@ void openLogFile()
+        ForEach(*queues)
+        {
+            IPTree &queue = queues->query();
+            const char *qname = queue.queryProp("@name");
+            newSchedulerQueues.emplace(qname);
+        }
+#else
+        const char *processName = config->queryProp("@name");
+        Owned<IStringIterator> targetClusters = getTargetClusters("EclSchedulerProcess", processName);
+        ForEach(*targetClusters)
+        {
+            SCMStringBuffer targetCluster;
+            targetClusters->str(targetCluster);
+            newSchedulerQueues.emplace(targetCluster.str());
+        }
+#endif
+        if (newSchedulerQueues.empty())
+            ERRLOG("No queues found to listen on");
+
+        CriticalBlock b(updateSchedulersCS);
+        if (newSchedulerQueues == schedulerQueues)
+        {
+            DBGLOG("Queue names have not changed, not updating");
+            return false;
+        }
+
+        StringBuffer oldNames, newNames;
+        toString(schedulerQueues, oldNames);
+        toString(newSchedulerQueues, newNames);
+        PROGLOG("Updating queue due to queue names change from '%s' to '%s'", oldNames.str(), newNames.str());
+
+        schedulerQueues = newSchedulerQueues;
+
+        return true;
+    }
+
+    void updateSchedulerQueues()
+    {
+        std::set<std::string> newQueueNames = schedulerQueues;
+        // stop and remove schedulers not in the new list as they are no longer needed
+        // for those in the new list, if they are already running, leave them alone
+        ForEachItemInRev(schedIdx, schedulers)
+        {
+            EclScheduler &scheduler = schedulers.item(schedIdx);
+            if (newQueueNames.find(scheduler.getServerName()) == newQueueNames.end())
+            {
+                // Queue has been removed
+                DBGLOG("Stopped listening to queue %s", scheduler.getServerName());
+                scheduler.stop();
+                schedulers.remove(schedIdx);
+            }
+            else
+            {
+                // "New" queue already running
+                newQueueNames.erase(scheduler.getServerName());
+            }
+        }
+
+        // only new schedulers will be left in the list
+        for (auto &newQueueName : newQueueNames)
+        {
+            DBGLOG("Start listening to queue %s", newQueueName.c_str());
+            Owned<EclScheduler> scheduler = new EclScheduler(this, newQueueName.c_str());
+            scheduler->start();
+            schedulers.append(*scheduler.getClear());
+        }
+
+        if (schedulers.empty())
+            ERRLOG("No clusters found to schedule for");
+    }
+
+    void stop()
+    {
+        DBGLOG("stop()");
+        CriticalBlock b(updateSchedulersCS);
+        ForEachItemIn(schedIdx, schedulers)
+        {
+            schedulers.item(schedIdx).stop();
+        }
+    }
+
+    void toString(const std::set<std::string> &queueNames, StringBuffer &queueNameList)
```

Comment:
If this function remained, EclSchedulerServer::toString would be a bit misleading as a name, and should probably be moved out of class as a utility function, but prob. not worth keeping this method (see other comments), since the update  method lists changes.